framework/decode/dx12_resource_value_mapper.cpp

```diff
@@ -1226,11 +1240,40 @@ void Dx12ResourceValueMapper::MapValue(const ResourceValueInfo& value_info,
         GFXRECON_CHECK_CONVERSION_DATA_LOSS(size_t, value_info.size);
         util::platform::MemoryCopy(&desc, sizeof(desc), desc_data_ptr, static_cast<size_t>(value_info.size));
         GetDispatchRaysResourceValues(indirect_values_map, value_info.state_object, desc);
+        return true;
+    }
+    else if (value_info.type == ResourceValueType::kExecuteIndirectCountBuffer)
+    {
+        uint32_t command_count = 0;
+
+        // Read command counts while CountBuffer contents is still visible in temp_resource_data
+        util::platform::MemoryCopy(&command_count,
+                                   sizeof(command_count),
+                                   temp_resource_data.data() + value_info.offset,
+                                   sizeof(command_count));
+        auto&    resource_value_infos = indirect_values_map[value_info.arg_buffer_extra_info.argument_buffer];
+        uint64_t command_byte_offset  = value_info.arg_buffer_extra_info.argument_buffer_offset;
+
+        // Insert new ArgumentBuffer RV infos, which will queue them for translation.
+        for (uint32_t i = 0; i < command_count; ++i)
+        {
+            for (const auto& resource_value_info :
+                 value_info.arg_buffer_extra_info.command_signature_info->resource_value_infos)
+            {
+                resource_value_infos.insert({ resource_value_info.offset + command_byte_offset,
+                                              resource_value_info.type,
+                                              resource_value_info.size,
+                                              value_info.state_object });
+            }
+            command_byte_offset += value_info.arg_buffer_extra_info.command_signature_info->byte_stride;
+        }
```

Comment:
This code block is similar to the else block in `PostProcessExecuteIndirect`. Could it be refactored into a function such as `GetExecuteIndirectResourceValues(ResourceValueInfoMap& resource_value_info_map, uint32_t command_count, etc..)`?

See `GetDispatchRaysResourceValues` which is a helper for getting resource-values for `ResourceValueType::kIndirectArgumentDispatchRays` and `Dx12ResourceValueMapper::PostProcessDispatchRays()`